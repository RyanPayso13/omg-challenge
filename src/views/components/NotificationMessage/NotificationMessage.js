import React from 'react';
import { Flex, Text, Button } from 'rebass';
import { OverlayWrap } from '../Styled';

const NotificationMessage = ({ message = '', callback, msgColor = "#627381" }) => {
    return (
        <OverlayWrap
            data-testid="notification-container">
            <Flex
                flexDirection="column"
                alignItems="center"
                justifyContent="center"
                width="100%">
                <Text 
                    data-testid="notification-message"
                    color={ msgColor }
                    mb={ 2 }>
                    { message }
                </Text>
                {!!callback && 
                    <Button 
                        data-testid="notification-cta"
                        border="1px solid"
                        borderColor="#2b6cb0"
                        color="#2b6cb0"
                        bg="white"
                        onClick={ callback }>
                        Close
                    </Button>
                }
            </Flex>
        </OverlayWrap>
    );
};

export default NotificationMessage;