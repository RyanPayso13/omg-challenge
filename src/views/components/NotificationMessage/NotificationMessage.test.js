import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import NotificationMessage from './NotificationMessage';

describe('<NotificationMessage />', () => {

    it('should render', () => {
        const { container } = render(<NotificationMessage />);
        expect(container.firstChild).toMatchSnapshot();    
    });

    it('should have a message', () => {
        const { getByTestId } = render(<NotificationMessage message="Message" />);
        expect(getByTestId('notification-message')).toHaveTextContent('Message'); 
    });

    it('should be the default message color', () => {
        const { getByTestId } = render(<NotificationMessage />);
        expect(getByTestId('notification-message')).toHaveStyle('color: #B62420'); 
    });

    it('should have a close button', () => {
        const { getByTestId } = render(<NotificationMessage callback={ jest.fn() } />);
        const cta = getByTestId('notification-cta');
        expect(cta).toBeInTheDocument(); 
        expect(cta).toHaveTextContent('Close'); 
    });  
    
    it('should execute the callback prop', () => {
        const dispatch = jest.fn();
        const { getByTestId } = render(<NotificationMessage callback={ dispatch } />);
        fireEvent.click(getByTestId('notification-cta'));
        expect(dispatch).toHaveBeenCalled(); 
    });  

});
