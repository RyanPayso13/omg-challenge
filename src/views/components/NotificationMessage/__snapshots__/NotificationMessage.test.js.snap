// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotificationMessage /> should render 1`] = `
<div
  class="sc-gZMcBi eSpJzV"
  data-testid="notification-container"
>
  <div
    class="sc-bdVaJa sc-bwzfXH dJYYq"
    width="100%"
  >
    <div
      class="sc-bdVaJa sc-htpNat AOxAi"
      color="#627381"
      data-testid="notification-message"
    />
  </div>
</div>
`;
